plugins/costinsight/src/plugin.ts

```diff
@@ -0,0 +1,32 @@
+import { createApiFactory, createPlugin, createRoutableExtension, discoveryApiRef } from '@backstage/core-plugin-api';
+
+import { rootRouteRef } from './routes';
+import { costinsightApiRef } from './api';
+import { bigqueryClient } from './api/bigqueryClient';
+
+export const costinsightPlugin = createPlugin({
+  id: 'bigqueryapi',
+  apis: [
+    createApiFactory({
+      api: costinsightApiRef,
+      deps: {
+        discoveryApi: discoveryApiRef,
+      },
+      factory: ({ discoveryApi }) =>
+        new bigqueryClient({ discoveryApi }),
+
+    }),
+  ],
+  routes: {
+    root: rootRouteRef,
+  },
+});
+
+export const BgcostinsightPage = costinsightPlugin.provide(
+  createRoutableExtension({
+    name: 'BgcostinsightPage',
+    component: () =>
+      import('./components/ExampleComponent').then(m => m.ExampleComponent),
```

Comment:
Why is this using the example component?

Reply:
apologies for naming, was in plan.